geode-core/src/main/java/org/apache/geode/internal/security/IntegratedSecurityService.java

```diff
@@ -221,8 +221,8 @@ public void authorize(Resource resource, Operation operation, Target target, Str
   }
 
   @Override
-  public void authorize(Resource resource, Operation operation, String target, String key) {
-    authorize(new ResourcePermission(resource, operation, target, key));
+  public void authorize(Resource resource, Operation operation, String target, Object key) {
+    authorize(new ResourcePermission(resource, operation, target, key.toString()));
```

Comment:
you should only call toString if key is not null. I think this is the cause of the test failures in pr tests.

Reply:
Thanks